lib/src/equals_matcher.dart

```diff
@@ -214,44 +243,24 @@ class _DeepMatcher extends Matcher {
       }
     }
 
-    var description = StringDescription();
-
     // If we have recursed, show the expected value too; if not, expect() will
     // show it for us.
     if (depth > 0) {
-      description
-          .add('was ')
-          .addDescriptionOf(actual)
-          .add(' instead of ')
-          .addDescriptionOf(expected);
-      return [description.toString(), location];
-    }
-
-    // We're not adding any value to the actual value.
-    return ['', location];
-  }
-
-  String _match(expected, actual, Map matchState) {
-    var rp = _recursiveMatch(expected, actual, '', 0);
-    if (rp == null) return null;
-    String reason;
-    if (rp[0].isNotEmpty) {
-      if (rp[1].isNotEmpty) {
-        reason = '${rp[0]} at location ${rp[1]}';
-      } else {
-        reason = rp[0];
-      }
+      return _Mismatch(location, actual,
+          (description, verbose) => description.addDescriptionOf(expected),
+          instead: true);
     } else {
-      reason = '';
+      return _Mismatch(location, actual, null);
     }
-    // Cache the failure reason in the matchState.
-    addStateInfo(matchState, {'reason': reason});
-    return reason;
   }
 
   @override
-  bool matches(item, Map matchState) =>
-      _match(_expected, item, matchState) == null;
+  bool matches(Object actual, Map matchState) {
+    var mismatch = _recursiveMatch(_expected, actual, '', 0);
+    if (mismatch == null) return true;
+    addStateInfo(matchState, {'mismatch': mismatch});
```

Comment:
This is a detail that would be good to describe in the commit message.

Reply:
Done